src/overlays/actors/ovl_En_Heishi4/z_en_heishi4.c

```diff
@@ -21,35 +33,352 @@ const ActorInit En_Heishi4_InitVars = {
+        }
+    }
+    this->actionFunc = func_80A56B40;
+}
+
+void func_80A5673C(EnHeishi4* this, GlobalContext* globalCtx) {
+    if (gSaveContext.eventChkInf[4] & 0x20) {
+        osSyncPrintf(VT_FGCOL(YELLOW) " ☆☆☆☆☆ マスターソード祝入手！ ☆☆☆☆☆ \n" VT_RST);
+        Actor_Kill(&this->actor);
+    } else {
+        this->unk_284 = 0;
+        if (gSaveContext.eventChkInf[8] & 1) {
+            if (!(gSaveContext.infTable[6] & 0x1000)) {
+                f32 frames = SkelAnime_GetFrameCount(&D_0600C444);
+                SkelAnime_ChangeAnim(&this->skelAnime, &D_0600C444, 1.0f, 0.0f, (s16)frames, 0, -10.0f);
+                this->actor.textId = 0x7007;
+                this->unk_282 = 5;
+                this->unk_284 = 1;
+                osSyncPrintf(VT_FGCOL(YELLOW) " ☆☆☆☆☆ デモ開始！ ☆☆☆☆☆ \n" VT_RST);
+            } else {
+                this->actor.textId = 0x7008;
+                this->unk_282 = 6;
+                osSyncPrintf(VT_FGCOL(BLUE) " ☆☆☆☆☆ 返事なし ☆☆☆☆☆ \n" VT_RST);
+            }
+            this->actionFunc = &func_80A56874;
+        } else {
+            Actor_Kill(&this->actor);
+        }
+    }
+}
+
+void func_80A56874(EnHeishi4* this, GlobalContext* globalCtx) {
+    if (this->unk_284 != 0) {
+        SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    }
+    if (func_8002F194(&this->actor, globalCtx) != 0) {
+        if (this->unk_284 == 0) {
+            this->actionFunc = func_80A5673C;
+
+        } else {
+            this->actionFunc = func_80A56900;
+        }
+    } else {
+        func_8002F2CC(&this->actor, globalCtx, 100.0f);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Heishi4/func_80A56328.s")
+void func_80A56900(EnHeishi4* this, GlobalContext* globalCtx) {
+    f32 frames = SkelAnime_GetFrameCount(&D_0600C6C8);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Heishi4/func_80A563BC.s")
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_0600C6C8, 1.0f, 0.0f, (s16)frames, 0, -10.0f);
+    this->actionFunc = func_80A56994;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Heishi4/func_80A56544.s")
+void func_80A56994(EnHeishi4* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    func_80038290(globalCtx, &this->actor, &this->unk_260.x, &this->unk_266.x, this->actor.posRot2.pos);
+    if (this->unk_282 == func_8010BDBC(&globalCtx->msgCtx)) {
+        if (func_80106BC8(globalCtx) != 0) {
+            func_80106CCC(globalCtx);
+            gSaveContext.infTable[6] |= 0x1000;
+            func_8002DF54(globalCtx, NULL, 8);
+            this->actionFunc = func_80A56A50;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Heishi4/func_80A56614.s")
+void func_80A56A50(EnHeishi4* this, GlobalContext* globalCtx) {
+    f32 frames = SkelAnime_GetFrameCount(&D_0600C374);
+    this->unk_288 = frames;
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_0600C374, 1.0f, 0.0f, frames, 2, -10.0f);
```

Comment:
Any chance the temp can be avoided here?
```suggestion
    this->unk_288 = SkelAnime_GetFrameCount(&D_0600C374);
    SkelAnime_ChangeAnim(&this->skelAnime, &D_0600C374, 1.0f, 0.0f, this->unk_288, 2, -10.0f);
```